game/src/main/scala/io/aigar/game/EntityContainer.scala

```diff
@@ -19,19 +23,24 @@ trait EntityContainer {
     None
   }
 
-  def handleCollision(entities: List[Entity], players: List[Player]): List[Entity] ={
+  def handleCollision(entities: List[Entity],
+                      players: List[Player],
+                      scoreModifications: Option[MutableList[ScoreModification]]): List[Entity] ={
     var entitiesReturn = List[Entity]()
     for (entity <- entities){
       for (player <- players) {
         for (cell <- player.cells) {
           if (cell.contains(entity.position)) {
-            entitiesReturn :::= onCellCollision(cell, entity)
+            entitiesReturn :::= onCellCollision(cell, Some(player), entity, scoreModifications)
           }
         }
       }
     }
     entities diff entitiesReturn
   }
 
-  def onCellCollision(cell: Cell, entity: Entity): List[Entity]
+  def onCellCollision(cell: Cell,
+                      player: Option[Player],
```

Comment:
Why not `Player` instead of `Option[Player]`? I understand that we won't always need to use it, but I'm not sure I get why it's a problem to always pass it.